### PR TITLE
Adding Missing Interpolation Methods to SyntheticHistory and fixing Missing ARMA Cluster Feature

### PR DESCRIPTION
**Pull request by GabrielSoto-INL**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
#2162 
#2166 

##### What are the significant changes in functionality due to this change request?
Some missing methods have been added to `raven/ravenframework/SupervisedLearning/SyntheticHistory.py` to allow interpolation between missing macro steps (e.g., missing years). 

Also a small fix that enables clustering on variance when using TSA ARMA algorithm. 

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [x] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

File: tests/framework/ROM/TimeSeries/SyntheticHistory/TrainingData/Interpolated_A.csv [hunk 1]
```diff
@@ -0,0 +1,3 @@
+macro,scaling,filename
+0,1,Clustered_A_0.csv
+1,1,Clustered_A_0.csv
```
Comment (dylanjm) [inline tests/framework/ROM/TimeSeries/SyntheticHistory/TrainingData/Interpolated_A.csv:3]: Is this testing a skipped year? From `0` to `2` maybe?

**Review by dylanjm — CHANGES_REQUESTED** [review]
Small question before looking deeper:

File: tests/framework/ROM/TimeSeries/SyntheticHistory/TrainingData/Interpolated_A.csv (hunk 1 above)
Comment (GabrielSoto-INL) [reply tests/framework/ROM/TimeSeries/SyntheticHistory/TrainingData/Interpolated_A.csv:3]: It is not, thanks for catching that! I reverted it back to reproduce the error when I submitted the Github issue and forgot to change it back. Will fix now

File: ravenframework/SupervisedLearning/SyntheticHistory.py [hunk 2]
```diff
@@ -215,6 +215,68 @@ def findAlgoByName(self, name):
...
+    # NOTE: we deepcopy'd a ROM to get here... so any non-clusterable features have
+    #       been copied over. For example: ARMA 'results', 'model', 'initials'
+    # TODO: these attributes should be overloaded in some fashion in the future
+    self.setLocalRomClusterFeatures(features)
+    self.amITrained = True
+
+  def parametrizeGlobalRomFeatures(self, featureDict):
+    """
+      Parametrizes the GLOBAL features of the ROM (assumes this is the templateROM and segmentation is active)
+      @ In, featureDict, dictionary of features to parametrize
```
Comment (PaulTalbot-INL) [inline ravenframework/SupervisedLearning/SyntheticHistory.py:261]: missing type in docstring after name (I'm sure it's missing in the original, too, sorry)

File: ravenframework/SupervisedLearning/SyntheticHistory.py [hunk 3]
```diff
@@ -215,6 +215,68 @@ def findAlgoByName(self, name):
...
 
+  def getFundamentalFeatures(self, requestedFeatures, featureTemplate=None):
+    """
+      Collect the fundamental parameters for this ROM
+      Used for writing XML, interpolating, clustering, etc
+      NOTE: This is originally copied over from SupervisedLearning!
+         Primarily using this for interpolation.
+      @ In, requestedFeatures, dict(list), featureSet and features to collect (may be None)
+      @ In, featureTemplate, str, templated string for naming features (probably leave as None)
+      @ Out, features, dict,
```
Comment (PaulTalbot-INL) [inline ravenframework/SupervisedLearning/SyntheticHistory.py:226]: missing description

File: ravenframework/SupervisedLearning/SyntheticHistory.py [hunk 4]
```diff
@@ -215,6 +215,68 @@ def findAlgoByName(self, name):
...
     return None
 
+  def getFundamentalFeatures(self, requestedFeatures, featureTemplate=None):
+    """
+      Collect the fundamental parameters for this ROM
+      Used for writing XML, interpolating, clustering, etc
+      NOTE: This is originally copied over from SupervisedLearning!
+         Primarily using this for interpolation.
+      @ In, requestedFeatures, dict(list), featureSet and features to collect (may be None)
+      @ In, featureTemplate, str, templated string for naming features (probably leave as None)
```
Comment (PaulTalbot-INL) [inline ravenframework/SupervisedLearning/SyntheticHistory.py:225]: missing `optional` after type / before description

**Review by PaulTalbot-INL — CHANGES_REQUESTED** [review]
A couple minor changes, otherwise this all looks fine. I suspect we'll want to add global tools at some point, but I also suspect MRA is going to lead to better solutions for this regime splitting anyway.

**Review by dylanjm — APPROVED** [review]
Changes LGTM

**Pull request merged**